File: frontend/src/types/image.ts
export interface Image {
  id: number
  name: string
  version: string
  architecture: string
  registry_path: string
  description?: string
  created_at?: string
  updated_at?: string
  created_by_id: number
}

export interface ImageCreate {
  name: string
  registry_path: string
  description?: string
}

export interface ImageUpdate {
  name?: string
  registry_path?: string
  description?: string
} 

File: backend/app/schemas/image.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class ImageBase(BaseModel):
    name: str
    description: Optional[str] = None
    architecture: str
    registry_path: str

class ImageCreate(ImageBase):
    pass

class ImageUpdate(ImageBase):
    pass

class Image(ImageBase):
    id: int
    created_at: datetime
    created_by_id: int

    class Config:
        from_attributes = True 